File: MyRixi-API/Models/Post.cs
namespace MyRixiApi.Models;

public enum PostState
{
    Draft,
    Published,
    Archived
}

public enum PostType
{
    Text
}

public class Post
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime PublishedAt { get; set; } = DateTime.UtcNow;
    public PostState State { get; set; } = PostState.Draft;
    public PostType Type { get; set; } = PostType.Text;
    
    // Relationships
    public Guid CommunityId { get; set; }
    public Community Community { get; set; } = null!;
    public Guid CommunityProfileId { get; set; }
    public CommunityProfile CommunityProfile { get; set; } = null!;
    public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    public ICollection<Attachment> Attachments { get; set; } = new List<Attachment>();
    public virtual ICollection<Like> Likes { get; set; } = new HashSet<Like>();
    public ICollection<Tag> Tags { get; set; } = new List<Tag>();
}

public class Like
{
    public Guid Id { get; set; }
    public Guid PostId { get; set; }
    public Guid UserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public virtual Post Post { get; set; }
    public virtual User User { get; set; }
}

File: MyRixi-Mobile/types/post.ts
import { TagResponse } from './tag';
import { Attachment } from './attachment';
import { CommentResponse } from './comment';

export enum PostType {
  Standard = 0,
}

export enum PostState {
  Draft = 0,
  Published = 1,
}

export interface Author {
  id: string;
  username: string;
  displayName: string;
  profileImageUrl: string;
}

export interface Post {
  id: string;
  title: string;
  content: string;
  publishedAt: string;
  postType: PostType;
  state: PostState;
  userId: string;
  authorId: string;
  author: Author;
  communityId: string;
  communityName: string;
  tags: TagResponse[];
  attachments: Attachment[];
  comments: CommentResponse[];
  likesCount: number;
  commentsCount: number;
  isLiked: boolean;
}